boa_cli/src/helper.rs

```diff
@@ -124,10 +134,12 @@ impl Highlighter for LineHighlighter {
 
         let reg = Regex::new(
             r#"(?x)
-            (?P<identifier>[$A-z_]+[$A-z_0-9]*) |
+            (?P<identifier>\b[$A-z_]+[$A-z_0-9]*\b) |
             (?P<string_double_quote>"([^"\\]|\\.)*") |
             (?P<string_single_quote>'([^'\\]|\\.)*') |
-            (?P<op>[+\-/*%~^!&|=<>;:])"#,
+            (?P<template_literal>`([^`\\]|\\.)*`) |
+            (?P<op>[+\-/*%~^!&|=<>;:]) |
+            (?P<number>\b0[bB][01]+|0[xX][0-9a-fA-F]+|[0-9]+\.[0-9]*|[0-9]*\.[0-9]+|[0-9]+\b)"#,
```

Comment:
```suggestion
            (?P<number>0[bB][01](_?[01])*n?|0[oO][0-7](_?[0-7])*n?|0[xX][0-9a-fA-F](_?[0-9a-fA-F])*n?|(([0-9](_?[0-9])*\.([0-9](_?[0-9])*)?)|(([0-9](_?[0-9])*)?\.[0-9](_?[0-9])*)|([0-9](_?[0-9])*))([eE][+-]?[0-9](_?[0-9])*)?n?)"#,
```

Demo: https://regex101.com/r/xcg47j/1

Remove the word boundary and add support for:
- BigIntLiteralSuffix
- OctalInteger
- ExponentPart
- NumericalLiteralSeparator

Reply:
I like this, thanks! Why was the word boundary removed?

Reply:
I assumed your original one means `\b(0[bB][01]+|0[xX][0-9a-fA-F]+|[0-9]+\.[0-9]*|[0-9]*\.[0-9]+|[0-9]+)\b`.

It fails to match `.01` and `10.` which has '.'(dot) at the beginning or end of the number.

Reply:
I will check it locally just in case, since not using the word boundaries would sometimes highlight the numeric part of an identifier.